cpp/src/cider/exec/nextgen/operators/RowToColumnNode.cpp

```diff
@@ -96,39 +101,76 @@ class ColumnWriter {
+          // and will reallocate buffer if more capacity is needed
+          auto raw_data_buffer = context_.getJITFunction()->emitRuntimeFunctionCall(
+              "get_buffer_with_realloc_on_demand",
+              JITFunctionEmitDescriptor{
+                  .ret_type = JITTypeTag::POINTER,
+                  .ret_sub_type = JITTypeTag::INT8,
+                  .params_vector = {arrow_array_.get(),
+                                    actual_raw_length_buffer[index_].get(),
+                                    context_.getJITFunction()
+                                        ->createLiteral(JITTypeTag::INT32, 2)
+                                        .get()}});
+          auto actual_raw_data_buffer =
+              raw_data_buffer->castPointerSubType(JITTypeTag::INT8);
+          auto cur_pointer = actual_raw_data_buffer + actual_raw_length_buffer[index_];
+
+          context_.getJITFunction()->emitRuntimeFunctionCall(
+              "do_memcpy",
+              JITFunctionEmitDescriptor{.ret_type = JITTypeTag::VOID,
+                                        .params_vector = {cur_pointer.get(),
+                                                          values.getValue().get(),
+                                                          values.getLength().get()}});
+          // Save JITValues of output buffers to corresponding exprs.
+          buffers_.clear();
+          buffers_.append(
+              setNullBuffer(values.getNull()), raw_length_buffer, raw_data_buffer);
         })
         ->build();
+  }
 
-    // get latest pointer to data buffer, will allocate data buffer on first call,
-    // and will reallocate buffer if more capacity is needed
-    auto raw_data_buffer = context_.getJITFunction()->emitRuntimeFunctionCall(
-        "get_buffer_with_realloc_on_demand",
-        JITFunctionEmitDescriptor{
-            .ret_type = JITTypeTag::POINTER,
-            .ret_sub_type = JITTypeTag::INT8,
-            .params_vector = {
-                arrow_array_.get(),
-                actual_raw_length_buffer[index_].get(),
-                context_.getJITFunction()->createLiteral(JITTypeTag::INT32, 2).get()}});
-    auto actual_raw_data_buffer = raw_data_buffer->castPointerSubType(JITTypeTag::INT8);
-    auto cur_pointer = actual_raw_data_buffer + actual_raw_length_buffer[index_];
-
-    context_.getJITFunction()->emitRuntimeFunctionCall(
-        "do_memcpy",
-        JITFunctionEmitDescriptor{
-            .ret_type = JITTypeTag::VOID,
-            .params_vector = {
-                cur_pointer.get(), values.getValue().get(), values.getLength().get()}});
+  void writeVariableSizeTypeColForStrExpr() {
+    // Get values need to write
+    utils::FixSizeJITExprValue values(expr_->get_expr_value());
+
+    // we will borrow offset buffer to store intermiedate result(string ptr and
+    // length, 64 bits), so allocate (arrow_array_len_ * 8) bytes, actual offset size
+    // should be (arrow_array_len_ + 1) * 4 bytes
```

Comment:
Seems it's an intermediate and temporary buffer to store the string ptr and length, instead of previous `raw_length_buffer` and `raw_data_buffer`. I think you don't need above comments and rename `actual_raw_length_buffer` to `raw_str_ptr_buffer`?